src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java

```diff
@@ -42,19 +53,23 @@ public void createPrimaryUserAffiliation(String eventPayload) {
         return;
       }
 
-      boolean isPrimaryAffiliationExists = userTenantService
-        .checkUserIfHasPrimaryAffiliationByUserId(consortiaTenant.getConsortiumId(), userEvent.getUserDto().getId());
+      boolean isPrimaryAffiliationExists = userTenantService.checkUserIfHasPrimaryAffiliationByUserId(consortiaTenant.getConsortiumId(), userEvent.getUserDto().getId());
       if (isPrimaryAffiliationExists) {
         log.warn("Primary affiliation already exists for tenant/user: {}/{}", userEvent.getTenantId(), userEvent.getUserDto().getUsername());
         return;
       } else {
         userTenantService.createPrimaryUserTenantAffiliation(consortiaTenant.getConsortiumId(), consortiaTenant, userEvent.getUserDto().getId(), userEvent.getUserDto().getUsername());
+        if (!Objects.equals(currentTenantId, consortiaTenant.getId())) {
+          userTenantService.save(consortiaTenant.getConsortiumId(), createUserTenant(folioExecutionContext.getTenantId(), userEvent));
+        }
       }
 
       PrimaryAffiliationEvent affiliationEvent = createPrimaryAffiliationEvent(userEvent);
       String data = objectMapper.writeValueAsString(affiliationEvent);
 
-      kafkaService.send(KafkaService.Topic.CONSORTIUM_PRIMARY_AFFILIATION_CREATED, consortiaTenant.getConsortiumId().toString(), data);
+      try (var context = new FolioExecutionContextSetter(prepareContextForTenant(requestContext.getTenantId(), folioModuleMetadata, requestContext))) {
```

Comment:
we should remain using central tenant context, so we should send CONSORTIUM_PRIMARY_AFFILIATION_CREATED only for primary tenant where user was created, fo central tenant we should not